migration_guide.md

```diff
@@ -0,0 +1,147 @@
+# IoTHub Python SDK Migration Guide
+
+This guide details the migration plan to move from the IoTHub Python v1 code base to the new and improved v2 
+code base.  
+
+## Installing the IoTHub Python SDK
+
+- v1
+
+```Shell
+pip install azure-iothub-device-client
+
+```
+
+- v2
+
+```Shell
+pip install azure-iot-device
+```
+
+## Creating a device client
+
+When creating a device client on the V1 client the protocol was specified on in the constructor.  With the v2 SDK we are
+currently only supporting the MQTT protocol so it only requires to supply the connection string when you create the client.
+
+### Symmetric Key authentication
+
+- v1
+
+```Python
+    from iothub_client import IoTHubClient, IoTHubClientError, IoTHubTransportProvider, IoTHubClientResult
+    from iothub_client import IoTHubMessage, IoTHubMessageDispositionResult, IoTHubError, DeviceMethodReturnValue
+
+    client = IoTHubClient(connection_string, IoTHubTransportProvider.MQTT)
+```
+
+- v2
+
+```Python
+    from azure.iot.device.aio import IoTHubDeviceClient
+    from azure.iot.device import Message
+
+    client = IoTHubDeviceClient.create_from_connection_string(connection_string)
+    await device_client.connect()
+```
+
+### x.509 authentication
+
+For x.509 device the v1 SDK required the user to supply the certificates in a call to set_options.  Moving forward in the v2
+SDK, we only require for the user to call the create function with an x.509 object containing the path to the x.509 file and
+key file with the optional pass phrase if neccessary.
+
+- v1
+
+```Python
+    from iothub_client import IoTHubClient, IoTHubClientError, IoTHubTransportProvider, IoTHubClientResult
+    from iothub_client import IoTHubMessage, IoTHubMessageDispositionResult, IoTHubError, DeviceMethodReturnValue
+
+    client = IoTHubClient(connection_string, IoTHubTransportProvider.MQTT)
+    # Get the x.509 certificate information
+    client.set_option("x509certificate", X509_CERTIFICATE)
```

Comment:
>X509_CERTIFICATE [](start = 41, length = 16)

Not sure if this is directly analogous across V1/V2

This seems to be some constant defined elsewhere in the file this snippet was taken from in the V1 section.

In the V2 section the values come from environment variables.

Reply:
I'm less concerned about the details of the v1 since if they're here they know how to do the v1 way.  Now I'm hoping that the v2 sample is self explanatory, but if not we may have to have some text at the beginning explaining the difference.  

---
In reply to: [333186880](https://github.com/Azure/azure-iot-sdk-python/pull/341#discussion_r333186880) [](ancestors = 333186880)